module/mfrc522_spi.c

```diff
@@ -17,22 +17,130 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
 	return byte;
 }
 
-u8 mfrc522_get_version(void)
+int mfrc522_get_version(void)
 {
-	return mfrc522_register_read(mfrc522_spi, MFRC522_VERSION_REG);
+	u8 version;
+	int ret;
+
+	ret = mfrc522_register_read(mfrc522_spi, MFRC522_VERSION_REG, &version,
+				    1);
+
+	if (ret < 0)
+		return ret;
+
+	return version;
+}
+
+void mfrc522_fifo_flush(void)
+{
+	u8 flush_byte = 1 << MFRC522_FIFO_LEVEL_REG_FLUSH_BIT;
```

Comment:
This is nitpicking but it would make more sense for the macro to be called `MFRC522_FIFO_LEVEL_REG_FLUSH_SHIFT`. Because the bit isn't actually worth 7, the shift is. Or define this macro as `1 << 7` and use `u8 flush_byte = MFRC522_FIFO_LEVEL_REG_FLUSH_BYTE`